packages/react/src/components/Menu/MenuItem.tsx

```diff
@@ -136,32 +155,19 @@ export const MenuItem = forwardRef<HTMLLIElement, MenuItemProps>(
         return;
       }
 
-      const { x, y, width, height } = menuItem.current.getBoundingClientRect();
-      if (rtl) {
-        setBoundaries({
-          x: [-x, x - width],
-          y: [y, y + height],
-        });
-      } else {
-        setBoundaries({
-          x: [x, x + width],
-          y: [y, y + height],
-        });
-      }
-
       setSubmenuOpen(true);
     }
 
     function closeSubmenu() {
       setSubmenuOpen(false);
-      setBoundaries({ x: -1, y: -1 });
     }
 
     function handleClick(
       e: KeyboardEvent<HTMLLIElement> | MouseEvent<HTMLLIElement>
     ) {
       if (!isDisabled) {
         if (hasChildren) {
+          setSubmenuOpen(true);
           openSubmenu();
```

Comment:
`setSubmenuOpen` is already called inside `openSubmenu`

```suggestion
          openSubmenu();
```